File: proposed-projects/round-trip-experiment/reverse/20260909T142951Z_4ff7cdec/20260910T082818Z_6d83d454/ported_implementation/gbnf/grammar_graph/get_input_as_code_points.py
from typing import List

from .grammar_graph_types import ValidInput


def get_code_point(char: str) -> int:
    if not char:
        raise ValueError(f"Could not get code point for character: {char}")
    return ord(char[0])


def get_input_as_code_points(src: ValidInput) -> List[int]:
    if isinstance(src, bool):
        raise ValueError(f"Invalid input type: {type(src).__name__}")

    if isinstance(src, int):
        return [src]

    if isinstance(src, list):
        for code_point in src:
            if not isinstance(code_point, int) or isinstance(code_point, bool):
                raise ValueError(
                    f"code_point must be an integer for {code_point} if src is a list"
                )
        return src

    if isinstance(src, str):
        return [get_code_point(char) for char in src]

    raise ValueError(f"Invalid input type: {type(src).__name__}")

File: packages/gbnf-experiment/data/20260909T151143Z_170b737d/ported_implementation/src/grammar_graph/get_input_as_code_points.ts
import type { ValidInput } from "./grammar_graph_types.ts";

export const get_code_point = (char: string): number => {
  const code_point = char.codePointAt(0);
  if (code_point === undefined) {
    throw new Error(`Could not get code point for character: ${char}`);
  }
  if (!Number.isInteger(code_point)) {
    throw new Error("code_point must be an integer!");
  }
  return code_point;
};

export const get_input_as_code_points = (src: ValidInput): number[] => {
  if (typeof src === "number") {
    return [src];
  }

  if (Array.isArray(src)) {
    for (const c of src) {
      if (!Number.isInteger(c)) {
        throw new Error(`code_point must be an integer for ${c} if src is a list`);
      }
    }
    return src;
  }

  if (typeof src === "string") {
    return [...src].map((s) => get_code_point(s));
  }

  throw new Error(`Invalid input type: ${typeof src}`);
};
